app/models/site.rb

```diff
@@ -36,5 +36,22 @@ def generate_png
 
   validates :user_id, presence: true
   validates :url, format: /\A((http|https):\/\/)*[a-z0-9_-]{1,}\.*[a-z0-9_-]{1,}\.[a-z]{2,5}(\/)?\S*\z/i
+  validate  :check_customer_status, on: :create
```

Comment:
Add custom validators for checking when a customer has cancelled or when a customer has exceeded their usage.

This will prevent customers from being able to create additional screen captures (they will either have to become an active customer, or move to a higher pricing tier)

Admins of course are exempt from this.